screeps-profiler.js

```diff
@@ -14,20 +14,36 @@ function setupProfiler() {
     profile(duration, filter) {
       setupMemory('profile', duration || 100, filter);
     },
+    background(duration, filter) {
+      setupMemory('background', duration || false, filter);
+    },
+    clear() {
+      var restart = !!Memory.profiler && Memory.profiler.type == 'background'
+      var filter = (restart && Memory.profiler.filter) ? Memory.profiler.filter : null
+      var duration = (restart && Memory.profiler.disableTick) ? Memory.profiler.disableTick - Game.time : null
+      resetMemory()
+      if(restart) {
+        setupMemory('background', duration, filter);
+      }
+    },
     reset: resetMemory,
+    output(numresults) {
+      return Profiler.output(numresults);
+    }
   };
 
   overloadCPUCalc();
 }
 
 function setupMemory(profileType, duration, filter) {
   resetMemory();
+  var disableTick = Number.isInteger(duration) ? Game.time + duration : false
```

Comment:
I'm not sure I follow how that would work- logic somewhere would have to catch that it's a background type and then increment it, so something would still need to change. I think by letting this have "false" as an option it opens up the door to other non-outputted types of profiling, but mostly I just don't see how that change is bigger than incrementing.